tools/tpm2_pkcs11/db.py

```diff
@@ -532,6 +532,23 @@ def _update_on_7(self, dbbakcon):
                     attrs[CKA_ALLOWED_MECHANISMS] = list(deduped_attrs)
                     Db._updatetertiary(dbbakcon, t['id'], attrs)
 
+    def _update_on_8(self, dbbakcon):
+        '''
+        Between version 7 and 8 of the DB the following changes need to be made:
+
+        Table tobjects:
+
+        The YAML attributes for a int sequence has to be a yaml sequence.
+        '''
+        c = dbbakcon.cursor()
+
+        c.execute('SELECT * from tobjects')
+        tobjs = c.fetchall()
+
+        for t in tobjs:
+            attrs = yaml.safe_load(io.StringIO(t['attrs']))
```

Comment:
does this actually work, if you had a V7 database? As t['attrs'] is either a list or a string, shouldn't the check be:
1. if string, break it apart based on sizeof the data type? You could use something `ctypes.sizeof(c_ulong)`, then take that result multiply it by two and split the hex strings on that and create a list, then convert that list to yaml update the attributes and then update the DB

Reply:
right, it was wrong ! hold on.

Reply:
@williamcroberts , is https://github.com/tpm2-software/tpm2-pkcs11/commit/f15726cf91de0d5f4612a957e981981b020f1835 what you'd like ?

Reply:
> @williamcroberts , is [f15726c](https://github.com/tpm2-software/tpm2-pkcs11/commit/f15726cf91de0d5f4612a957e981981b020f1835) what you'd like ?

Yeah that looks right, just some minor nits on that patch

Reply:
Which ones ? Which modifications would you like?